Storage/test.cc

```diff
@@ -1599,6 +1599,58 @@ TEST(TransactionalStorage, LastModifiedInMatrixContainers) {
   }
 }
 
+TEST(TransactionalStorage, WaitUntilLocalLogIsReplayed) {
+  current::time::ResetToZero();
+
+  using namespace transactional_storage_test;
+  using Storage = TestStorage<SherlockStreamPersister>;
+  using stream_t = typename Storage::persister_t::sherlock_t;
+
+  const std::string storage_file_name =
+      current::FileSystem::JoinPath(FLAGS_transactional_storage_test_tmpdir, "storage_data");
+  const auto storage_file_remover = current::FileSystem::ScopedRmFile(storage_file_name);
+  // Write mutation log.
+  {
+    Storage master_storage(storage_file_name);
+    master_storage.ReadWriteTransaction([](MutableFields<Storage> fields) {
+      fields.d.Add(Record{"one", 1});
+    }).Go();
+    master_storage.ReadWriteTransaction([](MutableFields<Storage> fields) {
+      fields.d.Add(Record{"two", 2});
+    }).Go();
+    master_storage.ReadWriteTransaction([](MutableFields<Storage> fields) {
+      fields.d.Add(Record{"three", 3});
+    }).Go();
+  }
+
+  // Test following storage.
+  {
+    // Create stream using previously written log.
+    stream_t stream(storage_file_name);
+    // Move data authority from the stream.
+    struct StreamPublisherOwner {
+      void AcceptPublisher(std::unique_ptr<stream_t::publisher_t>) {}
+    } stream_publisher_owner;
+    stream.MovePublisherTo(stream_publisher_owner);
+    // Spawn following storage.
+    Storage storage(stream);
+    // Wait until the whole log is replayed.
+    storage.WaitForTransactionsCount(storage.InternalExposeStream().InternalExposePersister().Size());
```

Comment:
Probably cleaner to add an `EXPECT_EQ(3u, storage.InternalExposeStream().InternalExposePersister().Size());` to line 1625 and use plain `3u ` here and on the next line.